pkg/types/credentials.go

```diff
@@ -146,7 +142,7 @@ func GenerateCredentials() (*Credentials, error) {
 }
 
 func (c *Credentials) TLSExists() bool {
-	return c.TLS != nil && *c.TLS != TLS{}
+	return c.TLS != nil && !(c.TLS.Certificate == "" && c.TLS.Key == "")
```

Comment:
change it to CertificateExists

Reply:
!(c.TLS.Certificate == "" && c.TLS.Key == "") --> (c.TLS.Certificate != "" || c.TLS.Key != "")
Shouldn't it be (c.TLS.Certificate != "" && c.TLS.Key != "")?
Unless I got something wrong here...

Reply:
you right they both can't be empty, good catch
c.TLS.Certificate != "" && c.TLS.Key != ""

Reply:
Why it is wrong? we did it on purpose(for the update case) : if c.CertificateExists() {
		err := c.validateTLS() 
and then in validate tls we check each field 

func (c *Credentials) validateTLS() error {
	if c.TLS.Certificate == "" || c.TLS.Key == "" {
		return errors.New("tls public certificate and key should be provided")
	}

	if c.TLS.UseSMCertificate {
		return errors.New("only one of the options could be set, SM provided credentials or tls")
	}
	_, err := tls.X509KeyPair([]byte(c.TLS.Certificate), []byte(c.TLS.Key))
	if err != nil {
		return errors.New("invalidate TLS configuration: " + err.Error())
	}

	return nil
}

Reply:
you right I forgot it followed by validate call, 
you wanted to assure here that update leaves inconsistent data ignore my comment

Reply:
agree, resolving